pkg/services/ngalert/eval/eval.go

```diff
@@ -155,11 +162,57 @@ func executeCondition(ctx AlertExecCtx, c *models.Condition, now time.Time, data
 		return ExecutionResults{Error: err}
 	}
 
+	// eval captures for the '__value__' label.
+	captures := []NumberValueCapture{}
+
+	captureVal := func(refID string, labels data.Labels, value *float64) {
+		captures = append(captures, NumberValueCapture{
+			Var:    refID,
+			Value:  value,
+			Labels: labels.Copy(),
+		})
+	}
+
 	for refID, res := range execResp.Responses {
-		if refID != c.Condition {
-			continue
+		// for each frame within each response, capture value if it is a number.
+		for _, frame := range res.Frames {
+			if len(frame.Fields) != 1 || frame.Fields[0].Type() != data.FieldTypeNullableFloat64 {
+				continue
+			}
+			var v *float64
+			if frame.Fields[0].Len() == 1 {
```

Comment:
is there a chance this could panic? Shouldn't we validate the lenght of `frame.Fields`?

Reply:
Is done in the continue above.